### ISSUE TITLE
isensee2017_model input parameters : n_labels = 4?

### ISSUE THREAD

**Issue by fragalassi** [ellisdg/3DUnetCNN#164]
Hello,
just to clarify within the inputs parameters to the model:

isensee2017_model(input_shape=(4,128,128,128), n_base_filters=16, depth=5, dropout_rate = 0.3,
                      n_segmentation_levels=3, n_labels=4, optimizer=Adam, initial_learning_rate=5e-4,
                      loss_function=weighted_dice_coefficient_loss, activation_name="sigmoid")

What you set as n_labels=4 is rather the number of channels? Thank you !

**Comment by csudage** [comment]
n_labels = 4 is the number of channels for the last layer.

**Comment by FrancyGalassi** [comment]
Thank you for your answer. I am still a bit confused. 

In the file with the config parameters :
config["labels"] = (1, 2, 4)  # the label numbers on the input image
config["n_labels"] = len(config["labels"])

Shall the parameter config["n_labels"]  be the same as n_labels in isensee2017_model inputs (which will thus be equal to 3) ? This would be my understanding. While the input_shape=(4,128,128,128) depends on the number fo input modalities (t1,t2,...). 

Many thanks!
Francesca

**Comment by csudage** [comment]
Your understanding is not wrong，the '4' of input_shape depend on the number for the modalities, n_laber=3 in the config parameters,but there is a  background channel in last layer ,so the last n_laber=3+1. This my understanding，i hope it can help you.

**Comment by nysuka** [comment]
hi @FrancyGalassi 
to your original question, what you see in isensee2017.py

`def isensee2017_model(input_shape=(4, 128, 128, 128), n_base_filters=16, depth=5, dropout_rate=0.3,
                      n_segmentation_levels=3, n_labels=4, optimizer=Adam, initial_learning_rate=5e-4,
                      loss_function=weighted_dice_coefficient_loss, activation_name="sigmoid"):`

where n_labels=4, that's just the default value as is all the other arguments with an "=" assigned. these values get overridden when called in train_isensee2017.py, 

`# instantiate new model
        model = isensee2017_model(input_shape=config["input_shape"], n_labels=config["n_labels"],
                                  initial_learning_rate=config["initial_learning_rate"],
                                  n_base_filters=config["n_base_filters"])`

n_labels is set as 3 here. n_labels is the number of different segmentations (core, enhancing and edema in this example). 

nb_channels is the number of modalities ( t1, t1ce, t2, flair in this example), which though defined, i dont believed is used.

n_channels in data.py is `n_channels = len(training_data_files[0]) - 1` where training_data_files consists of the 4 modalities plus truth file. so n_channels = 4.

**Comment by WrathofBhuvan11** [comment]
how to do prediction for some new input mha files? using isensee2017?

**Comment by DecentMakeover** [comment]
@csudage  Hey i have set input shape as (96,128,128,128), base_filters =96, 

but the model breaks , i have 24 labels to predict 

how should i change these parameters? any suggestions?

**Comment by csudage** [comment]
@DecentMakeover i think the number of the labels  too much, you can choice  the models with different paraments for every  label.

**Comment by DecentMakeover** [comment]
Hmmm, sorry for the late reply , i dont understand what you mean by choice the models?
